sdk/eventhub/Azure.Messaging.EventHubs/src/Consumer/CheckpointStartingPosition.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Azure.Messaging.EventHubs.Consumer
+{
+    /// <summary>
+    ///   The position in a partition's event stream to use when updating a checkpoint, indicates that an event processor should begin reading from the next event.
+    /// </summary>
+    ///
+    public struct CheckpointStartingPosition
+    {
+        /// <summary>
+        ///   The offset to associate with the checkpoint. If there is a <see cref="SequenceNumber"/> associated with this checkpoint, then this value will be used for
+        ///   informational metadata. If no <see cref="SequenceNumber"/> is associated, then this will be used for positioning when events are read.
+        /// </summary>
+        ///
+        public long? Offset { get; }
+
+        /// <summary>
+        ///   The replication segment to associate with the checkpoint. Used in conjunction with the sequence number if using a geo replication enabled Event Hubs namespace.
+        /// </summary>
+        ///
+        public string ReplicationSegment { get; }
+
+        /// <summary>
+        ///   The sequence number to associate with the checkpoint. If populated, it indicates that a processor should begin reading from the next event in the stream.
+        /// </summary>
+        ///
+        public long? SequenceNumber { get; }
```

Comment:
I added this basically to enable: https://github.com/Azure/azure-sdk-for-net/pull/38351#discussion_r1336346494 I don't know what we would set this value otherwise?
If we set it as long.MinValue then GetCheckpoint will return that as the event position.
I guess we could do something like:
- if no sequence number is passed in, set this as long.MinValue
- and then in the checkpointStore implementation, don't write to the blob unless sequence number is > long.MinValue

Reply:
I was going to say that we disallow numbers less than 0 as inputs, so the `long.MinValue` proposal seems like a good idea - but apparently, we do not validate.   So, yeah... I'm not sure that I have a good answer other than nullable or to make the assumption that we'll never see a real sequence number less than 0.    

Thoughts?

Reply:
I'm leaning towards the assume `long.MinValue` rather than forcing the nullable, personally.

Reply:
One thing I'm also thinking about is that the default value for sequence number is long.minValue - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs#L283
I need to think about it a little bit more, I think the long.MinValue is a good option, I'm just trying to understand the implications around the default and how it might interact, I guess.

I also noticed that long.MinValue is the default for offset and we enforce offset > long.MinValue in updateCheckpoint. So, I think that could mean that this approach would work.